cachito/workers/pkg_manager.py

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+import os
+import subprocess
+import tarfile
+import tempfile
+
+from cachito.errors import CachitoError
+from cachito.workers.config import get_worker_config
+
+
+log = logging.getLogger(__name__)
+
+
+def resolve_gomod_deps(archive_path):
+
+    with tempfile.TemporaryDirectory(prefix='cachito-') as temp_dir:
+        source_dir = _mount_app_src(archive_path, temp_dir)
+        env = {
+            'GOPATH': temp_dir,
+            'GO111MODULE': 'on',
+            'GOCACHE': temp_dir,
+            'GOPROXY': get_worker_config().athens_url,
+            'PATH': os.environ.get('PATH', ''),
+        }
+        cmd = ('go', 'list', '-m', 'all')
+
+        go_list = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                                   text=True, encoding='utf-8', env=env, cwd=source_dir)
+        output, error_output = go_list.communicate()
+        if go_list.returncode != 0:
+            log.error(
+                'Fetching gomod dependencies with "%s" failed with: %s',
+                ' '.join(cmd),
+                error_output,
+            )
+            raise CachitoError('Fetching gomod dependencies failed')
+
+        deps = []
+        for line in output.splitlines():
+            parts = line.split(' ')
+            if len(parts) == 1:
+                # This is the application itself, not a dependency
+                continue
+            if len(parts) > 2:
```

Comment:
How about `if len(parts) != 2:`?

Reply:
Actually, I like to separate the two use cases. In the case where there's only one part, we can just ignore, but we really want to warn if it unexpectedly has more than 2 parts.